modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/SortedIndexMvStorage.java

```diff
@@ -51,68 +50,47 @@ public interface SortedIndexMvStorage {
     int BACKWARDS = 1 << 2;
 
     /**
-     * The sole purpose of this class is to avoid massive refactoring while changing the original IndexRow.
+     * Returns the Index Descriptor of this storage.
      */
-    interface IndexRowEx {
-        /**
-         * Key-only binary row if index-only scan is supported, full binary row otherwise.
-         */
-        BinaryRow row();
+    SortedIndexDescriptor indexDescriptor();
 
-        /**
-         * Returns indexed column value.
-         *
-         * @param idx PK column index.
-         * @return Indexed column value.
-         */
-        Object value(int idx);
-    }
+    /**
+     * Returns a factory for creating index rows for this storage.
+     */
+    IndexRowSerializer indexRowSerializer();
 
-    boolean supportsBackwardsScan();
+    /**
+     * Returns a class deserializing index columns.
+     */
+    IndexRowDeserializer indexRowDeserializer();
```

Comment:
Again, we should discuss the decision of having separate serializer/deserializer in index. It's not clear, maybe we should remove them

Reply:
Yes, I also don't like them, they should be removed when indices get integrated with the storage

Reply:
Ok, I guess we'll do it in https://issues.apache.org/jira/browse/IGNITE-17310